Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PMLGalileanAlgorithm.cpp

```diff
@@ -0,0 +1,174 @@
+            const Real S_ck = S_ck_arr(i,j,k);
+            const Complex T2 = Theta2_arr(i,j,k);
+
+            // Update E
+            fields(i,j,k,Idx::Exy) = T2*(C*fields(i,j,k,Idx::Exy) + S_ck*c2*I*ky*Bz_old);
+            fields(i,j,k,Idx::Exz) = T2*(C*fields(i,j,k,Idx::Exz) - S_ck*c2*I*kz*By_old);
+            fields(i,j,k,Idx::Eyz) = T2*(C*fields(i,j,k,Idx::Eyz) + S_ck*c2*I*kz*Bx_old);
+            fields(i,j,k,Idx::Eyx) = T2*(C*fields(i,j,k,Idx::Eyx) - S_ck*c2*I*kx*Bz_old);
+            fields(i,j,k,Idx::Ezx) = T2*(C*fields(i,j,k,Idx::Ezx) + S_ck*c2*I*kx*By_old);
+            fields(i,j,k,Idx::Ezy) = T2*(C*fields(i,j,k,Idx::Ezy) - S_ck*c2*I*ky*Bx_old);
+            // Update B
+            fields(i,j,k,Idx::Bxy) = T2*(C*fields(i,j,k,Idx::Bxy) - S_ck*I*ky*Ez_old);
+            fields(i,j,k,Idx::Bxz) = T2*(C*fields(i,j,k,Idx::Bxz) + S_ck*I*kz*Ey_old);
+            fields(i,j,k,Idx::Byz) = T2*(C*fields(i,j,k,Idx::Byz) - S_ck*I*kz*Ex_old);
+            fields(i,j,k,Idx::Byx) = T2*(C*fields(i,j,k,Idx::Byx) + S_ck*I*kx*Ez_old);
+            fields(i,j,k,Idx::Bzx) = T2*(C*fields(i,j,k,Idx::Bzx) - S_ck*I*kx*Ey_old);
+            fields(i,j,k,Idx::Bzy) = T2*(C*fields(i,j,k,Idx::Bzy) + S_ck*I*ky*Ex_old);
+        });
+    }
+};
+
+void PMLGalileanAlgorithm::InitializeSpectralCoefficients (
+    const SpectralKSpace& spectral_kspace,
+    const amrex::DistributionMapping& dm,
+    const Array<Real, 3>& v_galilean,
+    const amrex::Real dt)
+{
+    const BoxArray& ba = spectral_kspace.spectralspace_ba;
+    // Fill them with the right values:
+    // Loop over boxes and allocate the corresponding coefficients
+    // for each box owned by the local MPI proc
+    for (MFIter mfi(ba, dm); mfi.isValid(); ++mfi){
+
+        const Box& bx = ba[mfi];
+
+        // Extract pointers for the k vectors
+        const Real* modified_kx = modified_kx_vec[mfi].dataPtr();
+#if (AMREX_SPACEDIM==3)
+        const Real* modified_ky = modified_ky_vec[mfi].dataPtr();
+#endif
+        const Real* modified_kz = modified_kz_vec[mfi].dataPtr();
+        // Extract arrays for the coefficients
+        Array4<Real> C = C_coef[mfi].array();
+        Array4<Real> S_ck = S_ck_coef[mfi].array();
+        Array4<Complex> Theta2 = Theta2_coef[mfi].array();
+        // Extract reals (for portability on GPU)
+        Real vx = v_galilean[0];
+        Real vy = v_galilean[1];
+        Real vz = v_galilean[2];
+
+        // Loop over indices within one box
+        ParallelFor(bx,
+        [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+        {
+            // Calculate norm of vector
+            const Real k_norm = std::sqrt(
+                std::pow(modified_kx[i], 2) +
+#if (AMREX_SPACEDIM==3)
+                std::pow(modified_ky[j], 2) +
+                std::pow(modified_kz[k], 2));
+#else
+                std::pow(modified_kz[j], 2));
+#endif
+
+            // Calculate coefficients
+            constexpr Real c = PhysConst::c;
+            const Complex I{0.,1.};
```

Comment:
Since C++14 you can make this [`constexpr` constructible](https://en.cppreference.com/w/cpp/numeric/complex/complex) (and the `amrex::GpuComplex<T>` [supports this as well](https://github.com/AMReX-Codes/amrex/blob/95995073ff23e4051330c9f96fd2d195d77b9468/Src/Base/AMReX_GpuComplex.H#L32)):
```suggestion
            constexpr Complex I{0.,1.};
```